### DISCUSSION TITLE
how to smooth the stress change

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [idaholab/moose#28915] [Q&A Modules: Solid mechanics]
Hi all, 
I am working on coupling the THM simulator and seismicity simulator to model the seismic events happening in the faulted-geothermal sites. To do so, i coupled our in-house moose-based code (correspounding for THM) with a code (correspounding for dynaimc rupture and seismicity).
the coupling parameter is stress tensor between the codes.
after the seismicty, the focal mechanism appears in the reactivated area. theoritically, it should be clear at the correspounding horizon, but as you can see in the figure below the the focal mechanism is not clear (symetric) as well as the stress change is not smooth enough.
![Präsentation1](https://github.com/user-attachments/assets/0a5aaefa-1d39-4d12-8e72-0594216e2785)

Figure above shows the stress change as elemental variable, so i changed it o nodal variable (figure below) but still  the smoothness of the stress is not good enough and the focal mechanism is not clear. 
![Bild1](https://github.com/user-attachments/assets/776ee904-2241-4d57-a1dd-0732fdc98be1)
  
BTW, the element size around around the focal mechanism is 1 m. 
how can i resolve these issues?
Thanks in advance, 
Rahim

**Comment by GiudGiud** [comment]
Hello

Should you be refining the mesh further in that location? The results look quite coarse. 

How did you assess convergence of the nonlinear solver? Did you modify any executioner parameter?
